pkg/rpc/server/server.go

```diff
@@ -1974,55 +1997,61 @@ func (s *Server) runScriptInVM(t trigger.Type, script []byte, contractScriptHash
 	if err != nil {
 		faultException = err.Error()
 	}
-	var (
-		registerSession result.OnNewSession
-		params          *result.InvocationParams
-	)
+	var registerSession func(sessionID string, item stackitem.Item, id int, finalize func()) uuid.UUID
 	if s.config.SessionEnabled {
-		registerSession = s.registerSession
-		if s.config.SessionBackedByMPT {
-			params = &result.InvocationParams{
-				Trigger:            t,
-				Script:             script,
-				ContractScriptHash: contractScriptHash,
-				Transaction:        tx,
-				NextBlockHeight:    ic.Block.Index,
+		registerSession = func(sessionID string, item stackitem.Item, stackIndex int, finalize func()) uuid.UUID {
+			iterID := uuid.New()
+			s.sessionsLock.Lock()
```

Comment:
I wouldn't do this directly here, accumulate all iterators first and then start timers/register session.

Reply:
Ah, looks like I forgot that it's only called during JSON marshaling and thus after the exit from the call handler. OK then.